### DISCUSSION TITLE
nvim-tree - [NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.

### DISCUSSION THREAD

**Discussion by emrebdr** [nvim-tree/nvim-tree.lua#3027] [Q&A]
Hello, I'm working on a large php project, approximately ~80k files. But 2 minutes after opening this project on neovim, I get this error.

[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
[NvimTree] File system watcher failed (EMFILE) for path $path, halting watcher.
...
...
...

`Error Image`
<img width="1242" alt="Screenshot 2024-12-14 at 17 37 25" src="https://github.com/user-attachments/assets/41aa5586-f7f2-4660-b7d4-d752a1818b30" />

I get this error for all files in the project.

I don't use many plugins in neovim, only these plugins are installed => telescope, harpoon, lsp, nvim-treesitter

solutions I have tried:
```
- ulimit -n unlimited unlimited
- launchctl maxfiles unlimited unlimited
- update parser and treesitter
```

config:
```
  {
  	“nvim-treesitter/nvim-treesitter”,
  	opts = {
  		ensure_installed = {
  			“vim”, ‘lua’, ‘vimdoc’, ”php”
  		},
  	},
  }
```

Do you have any idea why this is happening or a solution? thank you 🙏

**Comment by alex-courtis** [comment]
EMFILE normally happens on node creation and will report:
> fs.inotify.max_user_watches exceeded, see https://github.com/nvim-tree/nvim-tree.lua/wiki/Troubleshooting

This is the first time we've seen it on an event. I'll look into updating that error message.

Either way, the fix linked should work.

**Reply by alex-courtis** [reply]
#3028 added the message.

**Reply by emrebdr** [reply]
Heyy thank you for your answer, I tried the solution you suggested, but it didn't work and I still have the same problem, but the error message became a bit more detailed. That's why I added the screenshot to the question

**Reply by alex-courtis** [reply]
What did you set `fs.inotify.max_user_watches` and `fs.inotify.max_user_instances` to?
